velox/vector/DictionaryVector.h

```diff
@@ -214,6 +215,18 @@ class DictionaryVector : public SimpleVector<T> {
     setInternalState();
   }
 
+  void resize(vector_size_t size, bool setNotNull = true) override {
+    if (indices_ && indices_->refCount() > 1) {
+      VELOX_FAIL("Cannot resize shared indices.");
```

Comment:
Instead of failing, can we just allocate a new buffer and copy the data? Also, can we do that only if size > current size?